src/torchmetrics/collections.py

```diff
@@ -547,6 +547,10 @@ def __getitem__(self, key: str, copy_state: bool = True) -> Metric:
 
         """
         self._compute_groups_create_state_ref(copy_state)
+        if self.prefix is not None:
+            key = key.removeprefix(self.prefix)
+        if self.postfix is not None:
+            key = key.removesuffix(self.postfix)
```

Comment:
```suggestion
        if self.prefix:
            key = key.removeprefix(self.prefix)
        if self.postfix:
            key = key.removesuffix(self.postfix)
```
Could it be this simple?